loggercontext/logger_context.go

```diff
@@ -0,0 +1,48 @@
+// Package loggercontext stores and accesses the container struct in
+// context.Context.
+package loggercontext
+
+import (
+	"context"
+)
+
+// key is an unexported type for keys defined in this package. This prevents
+// collisions with keys defined in other packages.
+type key string
+
+// loggerKey is the key for logger struct values in context.Context. Clients use
+// loggercontext.NewContext and loggercontext.FromContext instead of using this
+// key directly.
+var loggerKey key = "logger"
+
+// Container is a communication structure used to transport information from one
+// resource to another. Container move between resource during reconciliation
+// within the dispatched context.
+type Container struct {
+	// KeyVals is a mapping of key-value pairs a micro logger adds to the log
+	// message issuance.
+	KeyVals map[string]string
+}
```

Comment:
Why not just `type KeyVals map[string]string`?

Reply:
No real reason. Was easier for me to copy/paste the message context from the kvm-operator and I took it that way. Maybe one day this will be beneficial.